src/apps/workspaces/crud/user_applet_access.py

```diff
@@ -671,20 +720,67 @@ async def get_workspace_managers(
             .where(
                 UserAppletAccessSchema.soft_exists(),
                 UserAppletAccessSchema.owner_id == owner_id,
-                UserAppletAccessSchema.role != Role.RESPONDENT,
                 has_access,
+                UserAppletAccessSchema.role != Role.RESPONDENT,
                 UserAppletAccessSchema.applet_id == applet_id if applet_id else True,
             )
-            .group_by(UserSchema.id, UserAppletAccessSchema.title)
+            .group_by(UserSchema.id)
+        ).cte("accepted_users")
+
+        query: Query = (
+            select(
+                func.coalesce(accepted_users.c.id, invited_users.c.id).label("id"),
+                func.coalesce(accepted_users.c.first_name, invited_users.c.first_name).label("first_name"),
+                func.coalesce(accepted_users.c.last_name, invited_users.c.last_name).label("last_name"),
+                func.coalesce(accepted_users.c.email_encrypted, invited_users.c.email).label("email_encrypted"),
+                func.coalesce(accepted_users.c.created_at, invited_users.c.created_at).label("created_at"),
+                func.coalesce(accepted_users.c.last_seen, invited_users.c.created_at).label("last_seen"),
+                func.coalesce(accepted_users.c.roles, invited_users.c.roles).label("roles"),
+                func.coalesce(accepted_users.c.applets, invited_users.c.applets, []).label("applets"),
+                func.coalesce(accepted_users.c.titles, invited_users.c.titles, [])
+                .cast(ARRAY(StringEncryptedType(Unicode, get_key)))
+                .label("titles"),
+                func.coalesce(invited_users.c.status, InvitationStatus.APPROVED).label("status"),
+                func.coalesce(accepted_users.c.is_pinned, False).label("is_pinned"),
+            )
+            .select_from(invited_users)
+            .join(accepted_users, accepted_users.c.id == invited_users.c.user_id, isouter=True, full=True)
         )
 
         if query_params.filters:
             query = query.where(*_AppletUsersFilter().get_clauses(**query_params.filters))
 
         # Get total count before ordering to evaluate eligibility for ordering by encrypted fields
-        coro_total = self._execute(select(count()).select_from(query.with_only_columns(UserSchema.id).subquery()))
+        coro_total = self._execute(
+            select(count()).select_from(query.with_only_columns(literal_column("status")).subquery())
+        )
         total = (await coro_total).scalar()
 
+        class _AppletManagersOrdering(Ordering):
```

Comment:
Is there any memory overhead to defining this new class here on every function invocation? I see that you're accessing variables from the outer function scope, but I wonder if these could just be passed in a constructor, and the class defined once elsewhere

Reply:
This is python way to do it

Reply:
Ok, no prob. If you're comfortable that it's negligible then I'll resolve